### ISSUE TITLE
How to parse maven plugin configuration to CLI command?

### ISSUE THREAD

**Issue by Shhad** [dependency-check/DependencyCheck#5632]
Hi,
I am creating OWASP dep check process and I want to standardize it to perform check on pieplines using dependency-check CLI, but I have encountered one problem.
Using dependency-check maven plugin and having below configuration:
```
<profile>
            <id>owasp</id>
            <build>
                <plugins>
                    <plugin>
                        <groupId>org.owasp</groupId>
                        <artifactId>dependency-check-maven</artifactId>
                        <version>8.2.1</version>
                        <configuration>
                            <failBuildOnCVSS>7</failBuildOnCVSS>
                            <suppressionFile>${project.basedir}/owasp-dependency-check-suppressions.xml</suppressionFile>
                            <scanSet>
                                <fileSet>
                                    <directory>${project.basedir}</directory>
                                    <includes>
                                        <include>pom.xml</include>
                                    </includes>
                                </fileSet>
                            </scanSet>
                        </configuration>
                        <executions>
                            <execution>
                                <goals>
                                    <goal>check</goal>
                                </goals>
                            </execution>
                        </executions>
                    </plugin>
                </plugins>
            </build>
        </profile>
```
, report tells me that I have some new vulnerabilities and its ok.

But using dependency-check CLI from command line like this:

`dependency-check/bin/dependency-check.sh --project "MY PORJECT" --scan './PROJECT_PATH/pom.xml' --failOnCVSS 7 -- suppression ./PROJECT_PATH/owasp-dependency-check-suppressions.xml`
report tells that there is no vulnerabilities at all, I'm not sure if it performs check anyway because all takes about 4 seconds.
If I switch CLI command paths to:
`dependency-check/bin/dependency-check.sh --project "MY PORJECT" --scan './PROJECT_PATH/' --failOnCVSS 7 -- suppression ./PROJECT_PATH/owasp-dependency-check-suppressions.xml`
, now I have got many vulnerabilities witch I didnt have while performing check by maven plugin.

So finally, my question is: how to CLI command will look like to have same output as in maven plugin configuration which I provided?
Thank you in advance.

**Comment by jeremylong** [comment]
We would need to know a lot more about the project structure, what type of maven project (JAR, WAR, etc.), do you include bower or other front end tools? What types of findings are showing up with the CLI that aren't reported by Maven? Are the findings from the CLI actually false positives?

The CLI and the Maven plugin may have a common core engine - how they detect dependencies is vastly different.

**Comment by jeremylong** [comment]
In general, we always suggest using the maven or gradle plugin instead of the CLI... I would not recommend changing.

**Comment by Shhad** [comment]
Thanks for quick answer.

It's JAR project. In fact I didn't check all vulnerabilities if they actually are false positives.
I found that CLI sees doubled vulnerabilities, example:

```
my-project-1.0.0.jar: some-dependency-1.0.0.jar CVE-12345
...some other vulnerabilities...
my-project-1.0.0.jar: CVE-12345
```

So vulnerability CVE-12345 is in a specific dependency and in project summary?
I found this strange, because I did not see such behavior when I was using CLI in frontend project with package-lock.json and package.json files.

But as you suggested I will try to use maven plugin whenever it will be possible.
Thank you.
